### DISCUSSION TITLE
Network Activity Page

### DISCUSSION THREAD

**Discussion by cdnicoll** [noise-lab/iot-inspector-client#127] [Q&A]
<img width="1239" alt="image" src="https://user-images.githubusercontent.com/245761/168870572-387da57c-d998-49c5-98f5-3ec4445d6172.png">

1. http://127.0.0.1:53721/api/get_overall_bandwidth_consumption
2. http://127.0.0.1:53721/api/get_all_counterparties/86400
  - Expect API response to be counterparty names
```json
[
  'Adobe',
 'Synology'
 'Google' 
]
````
3. Combination of `http://127.0.0.1:53721/api/get_device_list` and `http://127.0.0.1:53721/api/get_overall_device_stats/DeviceA/86400`
  - We worry that this will be many API Calls
    - Look into graphql
    - webhooks?
  - Proposed Single API Call
```json
[
  {
    "device_id": "DeviceA",
    "user_device_name": "Amazon Echo",
    "ip_addr": "10.0.0.2",
    "total_inbound_byte_count": 823,
    "total_outbound_byte_count": 45,
    "counterparty_country_list": [
      "RU"
    ]
  },
  {
    "device_id": "DeviceB",
    "user_device_name": "Google Home",
    "ip_addr": "10.0.0.3",
    "total_inbound_byte_count": 823,
    "total_outbound_byte_count": 45,
    "counterparty_country_list": [
      "CN",
      "RU"
    ]
  }
]
```

4. http://127.0.0.1:53721/api/get_device_list

**Comment by crazyideas21** [comment]
GraphQL for sure. Maybe websocket?
